### PR TITLE
Dynamically create media routes (radio and tv) 

### PR DESCRIPTION
**Pull request by pjlee11**
Resolves https://github.com/bbc/simorgh-infrastructure/issues/607

**Overall change:** Dynamically generates the regexs needed to create the routes for all the radio and tv media pages. 

**Code changes:**

- Adds `src/app/routes/config/index.js` with the serviceId's that are whitelisted for each service
- Adds the function `buildMediaRoutes` which using the config described above creates an array of regexs per service, such as 
```
/:service(persian)/:serviceId(bbc_persian_radio|bbc_dari_radio)/:mediaId([a-z0-9]+):amp(.amp)?
```
- Passes an array for the `MediaPage` react router route [docs for passing as an array](https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/Route.md#path-string--string)
- Updates `mediaIdRegex` to be any alphanumeric (need for on-demand pids)
- Adds `mediaRadioRegexArray` to the Express config using spread notation

**Testing notes**
In the format:
http://localhost.bbc.com:7080/${service}/${whiteListedServiceId}/{$anyAlphaNumericCharacter}

With `service` and `whitelistedServiceId` coming from the config file `src/app/routes/config/index.js` we should manually check a variety of the pages render.

The following URLs should return a valid page:
```
http://localhost.bbc.com:7080/persian/bbc_dari_radio/liveradio
http://localhost.bbc.com:7080/persian/bbc_persian_radio/liveradio
http://localhost.bbc.com:7080/persian/bbc_persian_radio/gfdnkj4785
http://localhost.bbc.com:7080/indonesia/bbc_indonesian_radio/gfdnkj4785
http://localhost.bbc.com:7080/afrique/bbc_afrique_tv/livetv
http://localhost.bbc.com:7080/kyrgyz/bbc_kyrgyz_tv/gfdnkj4785
http://localhost.bbc.com:7080/afrique/bbc_afrique_tv/livetv.amp
http://localhost.bbc.com:7080/kyrgyz/bbc_kyrgyz_tv/gfdnkj4785.amp
``` 

The following URLs should not return a valid page: 
```
http://localhost.bbc.com:7080/persian/bbc_hausa_radio/liveradio
http://localhost.bbc.com:7080/foobar/bbc_hausa_radio/ghjd8464hf
http://localhost.bbc.com:7080/arabic/foo_bar/liveradio
http://localhost.bbc.com:7080/arabic/foo_bar
http://localhost.bbc.com:7080/afrique/bbc_kyrgyz_tv/livetv
http://localhost.bbc.com:7080/kyrgyz/bbc_afrique_tv/1234abcd
http://localhost.bbc.com:7080/afrique/bbc_kyrgyz_tv/livetv.amp
http://localhost.bbc.com:7080/kyrgyz/bbc_afrique_tv/1234abcd.amp
```

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

**Comment by ryanmccombe** [comment]
Looks great :). Only concern I have would be performance - that's a lot of regexes for react router to test on each request, but surely fine, especially if short-circuits early

**Comment by pjlee11** [comment]
Yeah it's quite a bit for react router to handle, I believe in the future we might have a layer between mozart/simorgh to handle routing as it becomes more complex

**Comment by pjlee11** [comment]
image

File: src/app/routes/index.js [hunk 1]
```diff
@@ -4,7 +4,11 @@ import MediaPage from '../containers/MediaPage';
 import getArticleInitialData from './getInitialData/article';
 import getFrontpageInitialData from './getInitialData/frontpage';
 import getMediaPageInitialData from './getInitialData/mediapage';
-import { articleRegexPath, frontpageRegexPath, mediaRegexPath } from './regex';
+import {
+  articleRegexPath,
+  frontpageRegexPath,
+  mediaRadioRegexArray,
```
Comment (dr3) [inline src/app/routes/index.js]:
```suggestion
  mediaRadioRegexPathsArray,
```

**Review by dr3 — APPROVED** [review]

File: src/app/routes/index.js (hunk 1 above)
Comment (dr3) [inline src/app/routes/index.js]: Missed TV

**Review by dr3 — CHANGES_REQUESTED** [review]

File: src/app/routes/index.js (hunk 1 above)
Comment (pjlee11) [reply src/app/routes/index.js]: This made me realise that I should combine the radio and tv output into a single array as they for the immediate future will share Component and params

**Review by dr3 — APPROVED** [review]

File: src/app/routes/config/index.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  afrique: ['bbc_afrique_radio', 'bbc_afrique_tv'],
+  amharic: ['bbc_amharic_radio'],
+  arabic: ['bbc_arabic_radio'],
+  bengali: ['bbc_bangla_radio'],
+  burmese: ['bbc_burmese_radio', 'bbc_burmese_tv'],
+  gahuza: ['bbc_gahuza_radio'],
+  gujarati: ['bbc_gujarati_tv'],
+  hausa: ['bbc_hausa_radio', 'bbc_hausa_tv'],
+  hindi: ['bbc_hindi_radio', 'bbc_hindi_tv'],
+  igbo: ['bbc_igbo_radio'],
```
Comment (pjlee11) [inline src/app/routes/config/index.js]:
```suggestion
```

**Review by jroebu14 — APPROVED** [review]
LGTM

**Comment by 12** [comment]
@pjlee11 Are we comfortable accepting that there's going to be no maximum length on `mediaId`?

**Comment by dr3** [comment]
@rhell Good shout, not a huge thing but if we can limit it to a smaller & restricted string in simorgh & mozart it would be ideal

**Comment by pjlee11** [comment]
@rhell the id can be "any string", we are currently limiting it to any alphanumeric. I've been told we "might" be safe to put a limit of up to 50 chars but thats still 36 to the power of 50 (`6.5331862e+77`) which is still a ridiculous amount of permutations.

The limitation will happen in Mozart

**Comment by paruchurisilpa** [comment]
Looks good to me..

**Pull request merged**